game_patch/object/event.cpp

```diff
@@ -339,8 +331,104 @@ ConsoleCommand2 debug_event_msg_cmd{
     }
 };
 
+CodeInjection trigger_activate_linked_objects_activate_event_in_multi_patch{
+    0x004C038C,
+    [] (auto& regs) {
+        if (rf::level.version >= 300) {
+            const rf::Object* const obj = regs.esi;
+            const rf::Event* const event = static_cast<const rf::Event*>(obj);
+            static const std::unordered_set blocked_event_types{{
+                rf::EventType::Load_Level,
+                rf::EventType::Drop_Point_Marker,
+                rf::EventType::Go_Undercover,
+                rf::EventType::Win_PS2_Demo,
+                rf::EventType::Endgame,
+                rf::EventType::Defuse_Nuke,
+                rf::EventType::Play_Video,
+            }};
+            // RF does not allow triggers to activate events in multiplayer.
+            // In AF levels, we allow it, unless it is an event that would be 
+            // problematic in multiplayer. 
+            const rf::EventType event_type = static_cast<rf::EventType>(event->event_type);
+            if (!blocked_event_types.contains(event_type)) {
+                // Allow activation.
+                regs.eip = 0x004C03C2;
+            } else {
+                // Do not allow activation.
+                regs.eip = 0x004C03D3;
+            }
+        }
+    }
+};
+
+CodeInjection EventUnhide__process_patch{
+    0x004BCDF0,
+    [] (const auto& regs) {
+        rf::Event* const event = regs.ecx;
+        rf::Event__process(event);
+    }
+};
+
+CodeInjection EventMakeInvulnerable__process_patch{
+    0x004BC8F0,
+    [] (const auto& regs) {
+        rf::Event* const event = regs.ecx;
+        rf::Event__process(event);
+    }
+};
+
+// These events are not implemented.
+bool af_is_forward_exempt(const rf::EventType event_type) {
```

Comment:
not sure what's the point of this code? You make a list of unsupported events to not forward events, because we don't know if they should be forwarded? If so then maybe we should disable it for any unsupported event, without making a list that will get outdated very fast

Reply:
No. They should not be forwarded. I included this code so if we support these events in future we do not have to worry about this function.